client/ts/src/uiWrapperObj.ts

```diff
@@ -0,0 +1,350 @@
+  /** Type of order (Limit, PostOnly, ...). */
+  orderType: OrderType;
+}
+
+export interface OpenOrderInternal {
+  price: Uint8Array;
+  clientOrderId: bignum;
+  orderSequenceNumber: bignum;
+  numBaseAtoms: bignum;
+  dataIndex: number;
+  lastValidSlot: number;
+  isBid: boolean;
+  orderType: number;
+  padding: bignum[]; // 30 bytes
+}
+
+/**
+ * Wrapper object used for reading data from a wrapper for manifest markets.
+ */
+export class UiWrapper {
+  /** Public key for the market account. */
+  address: PublicKey;
+  /** Deserialized data. */
+  private data: WrapperData;
+
+  /**
+   * Constructs a Wrapper object.
+   *
+   * @param address The `PublicKey` of the wrapper account
+   * @param data Deserialized wrapper data
+   */
+  private constructor({
+    address,
+    data,
+  }: {
+    address: PublicKey;
+    data: WrapperData;
+  }) {
+    this.address = address;
+    this.data = data;
+  }
+
+  /**
+   * Returns a `Wrapper` for a given address, a data buffer
+   *
+   * @param marketAddress The `PublicKey` of the wrapper account
+   * @param buffer The buffer holding the wrapper account data
+   */
+  static loadFromBuffer({
+    address,
+    buffer,
+  }: {
+    address: PublicKey;
+    buffer: Buffer;
+  }): UiWrapper {
+    const wrapperData = UiWrapper.deserializeWrapperBuffer(buffer);
+    return new UiWrapper({ address, data: wrapperData });
+  }
+
+  /**
+   * Updates the data in a Wrapper.
+   *
+   * @param connection The Solana `Connection` object
+   */
+  public async reload(connection: Connection): Promise<void> {
+    const buffer = await connection
+      .getAccountInfo(this.address)
+      .then((accountInfo) => accountInfo?.data);
+    if (buffer === undefined) {
+      throw new Error(`Failed to load ${this.address}`);
+    }
+    this.data = UiWrapper.deserializeWrapperBuffer(buffer);
+  }
+
+  /**
+   * Get the parsed market info from the wrapper.
+   *
+   * @param marketPk PublicKey for the market
+   *
+   * @return MarketInfoParsed
+   */
+  public marketInfoForMarket(marketPk: PublicKey): MarketInfoParsed | null {
+    const filtered: MarketInfoParsed[] = this.data.marketInfos.filter(
+      (marketInfo: MarketInfoParsed) => {
+        return marketInfo.market.toBase58() == marketPk.toBase58();
+      },
+    );
+    if (filtered.length == 0) {
+      return null;
+    }
+    return filtered[0];
+  }
+
+  /**
+   * Get the open orders from the wrapper.
+   *
+   * @param marketPk PublicKey for the market
+   *
+   * @return OpenOrder[]
+   */
+  public openOrdersForMarket(marketPk: PublicKey): OpenOrder[] | null {
+    const filtered: MarketInfoParsed[] = this.data.marketInfos.filter(
+      (marketInfo: MarketInfoParsed) => {
+        return marketInfo.market.toBase58() == marketPk.toBase58();
+      },
+    );
+    if (filtered.length == 0) {
+      return null;
+    }
+    return filtered[0].orders;
+  }
+
+  // Do not include getters for the balances because those can be retrieved from
```

Comment:
What if there are a bunch of markets and infrequent orders? Would rather not do a bunch of fetches when alright with stale data. I dont feel strongly about this though